src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.ValueTaskSource.cs

```diff
@@ -120,26 +125,19 @@ internal virtual void ReleaseNativeResource()
                     _overlapped = null;
                 }
 
-                // Ensure we're no longer set as the current AwaitableProvider (we may not have been to begin with).
+                // Ensure we're no longer set as the current ValueTaskSource (we may not have been to begin with).
                 // Only one operation at a time is eligible to use the preallocated overlapped
-                _strategy.CompareExchangeCurrentOverlappedOwner(null, this);
+                _strategy.TryToReuse(this);
             }
 
             private static void IOCallback(uint errorCode, uint numBytes, NativeOverlapped* pOverlapped)
             {
-                // Extract the AwaitableProvider from the overlapped.  The state in the overlapped
+                // Extract the ValueTaskSource from the overlapped.  The state in the overlapped
                 // will either be a AsyncWindowsFileStreamStrategy (in the case where the preallocated overlapped was used),
                 // in which case the operation being completed is its _currentOverlappedOwner, or it'll
-                // be directly the AwaitableProvider that's completing (in the case where the preallocated
+                // be directly the ValueTaskSource that's completing (in the case where the preallocated
                 // overlapped was already in use by another operation).
```

Comment:
This comment looks like it's now out of date, since the line below is always casting to ValueTaskSource.